state/application.go

```diff
@@ -3124,6 +3132,83 @@ func allUnits(st *State, application string) (units []*Unit, err error) {
 	return units, nil
 }
 
+// GetUnitAttachmentInfos returns a list of UnitAttachmentInfo for this application.
```

Comment:
We can expand this comment to explain that it is used to get info on storage attachments for units yet to be provisioned etc (see also comment below).